robotiq_c_model_control/src/robotiq_c_model_control/c_model_ethercat_node.cpp

```diff
@@ -0,0 +1,81 @@
+#include "ros/ros.h"
+
+#include <boost/bind.hpp>
+#include <boost/ref.hpp>
+
+#include "robotiq_c_model_control/c_model_ethercat_client.h"
+#include <robotiq_c_model_control/CModel_robot_input.h>
+#include "robotiq_ethercat/ethercat_manager.h"
+
+
+/*
+  Note that this code currently works only to control ONE CModel gripper
+  attached to ONE network interface. If you want to add more grippers
+  to the same network, you'll need to edit the source file.
+*/
+
+// Note that you will likely need to run the following on your binary:
+// sudo setcap cap_net_raw+ep <filename>
+
+
+void changeCallback(robotiq_c_model_control::CModelEtherCatClient& client,
+                    const robotiq_c_model_control::CModelEtherCatClient::GripperOutput::ConstPtr& msg)
+{
+  client.writeOutputs(*msg);
+}
+
+
+int main(int argc, char** argv)
+{
+  using robotiq_ethercat::EtherCatManager;
+  using robotiq_c_model_control::CModelEtherCatClient;
+
+  typedef CModelEtherCatClient::GripperOutput GripperOutput;
+  typedef CModelEtherCatClient::GripperInput GripperInput;
+
+  ros::init(argc, argv, "robotiqCModel");
```

Comment:
The node name is rarely used.  Launch files typically rename them.  It's safe to change it and good to adhere to naming conventions.